src/common/src/types/interval.rs

```diff
@@ -119,6 +105,63 @@ impl IntervalUnit {
         })
     }
 
+    /// Internal utility used by [`Self::mul_float`] and [`Self::div_float`] to adjust fractional
+    /// units. Not intended as general constructor.
+    fn from_floats(months: f64, days: f64, usecs: f64) -> Option<Self> {
+        // TSROUND in include/datatype/timestamp.h
+        // round eagerly at usecs precision because floats are imprecise
+        // should round to even #5576
+        let months_round_usecs =
+            |months: f64| (months * (USECS_PER_MONTH as f64)).round() / (USECS_PER_MONTH as f64);
+
+        let days_round_usecs =
+            |days: f64| (days * (USECS_PER_DAY as f64)).round() / (USECS_PER_DAY as f64);
```

Comment:
To make sure I understand this, let me check the logic:
1. Multiply # of months by usecs.
2. Round it.
3. Divide by it again.

The point of it is for rounding to occur at USECS precision, rather than at months precision, to make it more precise.

Reply:
The point of it is for rounding to occur at USECS precision, rather than unrounded. Rounding at months precision is not an option, because we do want to convert fractional months into days (and usecs).

Now comparing unrounded vs rounding to USECS precision. I encountered this example:
`(9 * 0.3).fract() * 30` is supposed to be `21`, but it evaluates to `20.999...` in practice. This makes a difference when we split the integer part and fractional part. So this implementation applies rounding before every call of `trunc_fract`.

Reply:
Makes sense, thanks!